### PR TITLE
Fetchpostwork

### PR DESCRIPTION
**Pull request by Dunn-Austen**
**What's this PR do?**
- Adds the foundation for the fetch-Post functionality into the bottom of the scripts.js file. This includes some event handlers as well as the fetch-post functions, most of which is commented out currently and not working properly. Some collaborative tweaks are needed
- Adds the HTML for the post functionality under the second column in our dashboard (I've made sure to use section (with class data-subsection) rather than article where applicable and have modeled things to fit Heather's recent semantic edits)
- Adds the accompanying Scss styling for the Post section's HTML (this is a rough draft and can use polishing to better match the overall tone of dashboard's styling - I'd built it before receiving these most recent styling updates)
- Removes the mixins for margin and padding and reverts the affected Css rules back to their original vanilla formats (per instructor suggestions on a former PR)
- Incorporates code from recent PRs by other collaborators, meaning that you may see edits reflected in the file changes that could summon deja-vu. 

**Where should the reviewer start?**
- Investigate styling contrasts involving the new "submit stats" section in the section column
- Eventually, we'll need to diagnose and solve the problem involving the dysfunctional post code

---
**Implications and Action Items**

### Miscellaneous
- The tabindex attribute on the new section may affect accessibility issues
- Styling may need to be adjusted for the new section

### Fetch-Post Work
- I haven't been able to fix the problems regarding the post functionality. My methodology involved commenting out the event listeners and post functions except for those involving sleepData (so that I could better isolate and problem solve), and then trying to run through the errors. I haven't had luck. 

**Rundown of the issues:**
- The post functions auto-run regardless of whether they are actually invoked. This might be a result of **asynchronous processing?** This makes the timing of the fetch firing a complicating factor.

- The **422 (Unprocessable Entity) status code** has repeatedly shown up. The following is an web excerpt: (Error 422) means the server understands the content type of the request entity (hence a 415(Unsupported Media Type) status code is inappropriate), and the syntax of the request entity is correct (thus a 400 (Bad Request) status code is inappropriate) but was unable to process the contained instructions.
For example, this error condition may occur if an XML request body contains well-formed (i.e., syntactically correct), but semantically erroneous, XML instructions.

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -64,6 +64,50 @@ <h3 class="subtext secondary-heading">USER WITH MOST SLEEP THIS WEEK</h3>
...
+                <input class="minutes-active" type="number" step="any" min="0" name="" value="" placeholder="Enter Numerical Data" required>
+              </div>
+              <div class="input-container">
+                <label for="">Flights of Stairs</label>
+                <input class="flights-stairs" type="number" step="any" min="0" name="" value="" placeholder="Enter Numerical Data" required>
+              </div>
+              <button class="activity__submit--btn" type="submit">Submit Data</button>
+            </form>
+          </section>
+        </section>
```
Comment (NaomiKC92) [inline src/index.html]: Semantic naming looks great!

File: src/scripts.js [hunk 2]
```diff
@@ -330,7 +323,100 @@ fetchData('users/userData')
...
   })
   .catch(error => console.log('hydrationData error'));
-
 })
+
 .catch(error => console.log('userData error'));
 // *****END OF .THEN FOR USERDATA**************************
+
+// Event listeners for show post-data form buttons
+$('.show__activity--btn').on('click', function() {
```
Comment (NaomiKC92) [inline src/scripts.js]: When we get this figured out, we should try to aim to change these to named functions that we would call here

File: src/index.html [hunk 3]
```diff
@@ -76,33 +76,33 @@ <h3 class="subtext">Input your stats</h3>
             <form class="sleep-form" action="index.html" method="post">
               <div class="input-container">
                 <label for="">Hours Slept</label>
-                <input class="hours-slept" type="number" step="any" min="0" name="" value="" placeholder="Enter Numerical Data" required>
+                <input class="hours-slept" type="number" step="any" min="0" name="hours-slept" value="" placeholder="Enter Numerical Data" required>
```
Comment (hfaerber) [inline src/index.html]: All changes in index.html are just adding the class name as the name attribute value as well.

File: src/scss/styles.scss [hunk 4]
```diff
@@ -31,22 +32,23 @@ canvas {
...
+  color: $body-background-color;
   @include font-look(2.2em, 600);
 }
 
 .welcome-message {
   border-radius: 3px;
-  @include margin(15px, 3px, 10px, 3px);
-  @include padding(3px, 15px, 3px, 20px);
+  margin: 15px, 3px, 10px, 3px);
+  padding: 3px 15px 3px 45px;
```
[the remark this answers is not in the document]
Comment (hfaerber) [reply src/scss/styles.scss]: I forgot to change the padding on `.welcome-message` back to 20px.

File: src/scss/styles.scss [hunk 5]
```diff
@@ -276,6 +275,7 @@ input {
     &:valid {
     border: 3px solid green;
     }
+}
```
Comment (hfaerber) [inline src/scss/styles.scss]: Just adding a curly brace that got lost in a merge conflict.

File: src/scripts.js [hunk 6]
```diff
@@ -348,75 +348,72 @@ $('.show__sleep--btn').on('click', function() {
...
-//           'Content-Type': 'application/json'
-//         },
-//       body: JSON.stringify(hydrationBody)
-//     })
-//       .then(response => response.json())
-//       .then(data => console.log(data))
-//       .catch(error => console.log('error'))
-//   }
-// }
+$('.sleep__submit--btn').on('click', postSleep);
```
Comment (hfaerber) [inline src/scripts.js]: This looks like a lot of edits but it was just uncommenting out the existing code so we could test it bit-by-bit and making the minor changes we worked through together.

-Capitalizing 'POST'
-Changing single quotes to double quotes and vice versa
-Adding event.preventDefault and event arg  **Good looking out @Dunn-Austen!**
-Removing quotes from around the object keys

**Pull request merged**